src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -484,10 +518,21 @@ public void TestEnvironmentVariables_Environment_DataRoundTrips()
 
             string envVarResultKey = "";
             string envVarResultValue = "";
-            foreach(KeyValuePair<string, string> entry in psi.EnvironmentVariables)
+
+            // .NET Core uses a Dictionary<string, string>, but the full .NET Framework uses
+            // StringDictionary. These collections return different types from GetEnumerator.
```

Comment:
Seems like there ought to be a generic way somewhere to compare two IDictionary's.